### PR TITLE
SQF Lint Cleanup Pass

### PR DESCRIPTION
**Pull request by PabstMirror**
Did a quick pass with the SQF Lint tool. (https://github.com/LordGolias/sqf)

Most are just private cleanups which aren't a big deal, but it did find some actual errors.
- Fixed: getDoorTurrets (Not used, was broken, could drop?)
- Fixed: getMGRSdata (Was ignoring getMapData altitude)
- Fixed: throttledPublicVariable (Not used anywhere, should probably test if we use it in the future)
- Fixed: handleRespawn (Using bad var name, escort setVar would not be reset)
- Fixed dropNozzle
- magazineRepackProgress (magically not broken because progressbar uses same var name)
- readWeaponDataFromConfig (same thing, `_weapon` always existed from upper scope)

### CODE REVIEW

File: addons/advanced_throwing/XEH_postInit.sqf [hunk 1]
```diff
@@ -115,7 +115,7 @@ addMissionEventHandler ["Draw3D", { // Blue is predicted before throw, red is re
         drawIcon3D ["\a3\ui_f\data\gui\cfg\hints\icon_text\group_1_ca.paa", [0,0,1,1], _newTrajAGL, 1, 1, 0, "", 2];
     } forEach GVAR(predictedPath);
     {
-        _newTrajAGL = _x;
+        private _newTrajAGL = _x;
         drawIcon3D ["\a3\ui_f\data\gui\cfg\hints\icon_text\group_1_ca.paa", [1,0,0,1], _newTrajAGL, 1, 1, 0, "", 2];
```
Comment (CorruptedHeart) [inline addons/advanced_throwing/XEH_postInit.sqf]: Could probably just use `_x` here without `_newTrajAGL`

**Comment by jonpas** [comment]
Oh, commy also opened the PRs for getDoorTurret and publicVariable. :D

**Comment by commy2** [comment]
missed this 🤦‍♀️

**Comment by commy2** [comment]
This PR closes #5159 and #5160.

**Comment by LordGolias** [comment]
Glad to know this was so helpful! Please share relevant issues to the SQF parser github please.

File: tools/sqf_linter.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    all_warnings = 0
+    all_errors = 0
+    
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-m','--module', help='only search specified module addon folder', required=False, default=".")
+    args = parser.parse_args()
+ 
+    for root, dirnames, filenames in os.walk('../addons' + '/' + args.module):
+      for filename in fnmatch.filter(filenames, '*.sqf'):
+        sqf_list.append(os.path.join(root, filename))
```
Comment (jonpas) [inline tools/sqf_linter.py]: 2 space instead of 4 space indentation.

File: tools/sqf_linter.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    args = parser.parse_args()
+ 
+    for root, dirnames, filenames in os.walk('../addons' + '/' + args.module):
+        for filename in fnmatch.filter(filenames, '*.sqf'):
+            sqf_list.append(os.path.join(root, filename))
+        
+    for filename in sqf_list:
+        warnings, errors = analyze(filename)
+        all_warnings = all_warnings + warnings
+        all_errors = all_errors + errors
```
Comment (SilentSpike) [inline tools/sqf_linter.py]: `+=`

File: tools/sqf_linter.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+            print("{}:".format(filename))
+            writer.write('    [%d,%d]:%s\n' % (e.position[0], e.position[1] - 1, e.message))
+            return 0, -1
+
+        exceptions = sqf.analyzer.analyze(result).exceptions
+        if (exceptions): 
+            print("{}:".format(filename))
+            for e in exceptions:
+                if (e.message.startswith("error")):
+                    errors = errors + 1
```
Comment (SilentSpike) [inline tools/sqf_linter.py]: `+=`

File: tools/sqf_linter.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+                else:
+                    warnings = warnings + 1
```
Comment (SilentSpike) [inline tools/sqf_linter.py]: `+=`

File: tools/sqf_linter.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    parser.add_argument('-m','--module', help='only search specified module addon folder', required=False, default=".")
+    args = parser.parse_args()
+ 
+    for root, dirnames, filenames in os.walk('../addons' + '/' + args.module):
+        for filename in fnmatch.filter(filenames, '*.sqf'):
+            sqf_list.append(os.path.join(root, filename))
+        
+    for filename in sqf_list:
+        warnings, errors = analyze(filename)
+        all_warnings = all_warnings + warnings
```
Comment (SilentSpike) [inline tools/sqf_linter.py]: `+=`

File: tools/sqf_linter.py [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    warnings = 0
+    errors = 0
+    with open(filename, 'r') as file:
+        code = file.read()
+        try:
+            result = parse(code)
+        except SQFParserError as e:
+            print("{}:".format(filename))
+            writer.write('    [%d,%d]:%s\n' % (e.position[0], e.position[1] - 1, e.message))
+            return 0, -1
```
Comment (SilentSpike) [inline tools/sqf_linter.py]: What's the reason this returns a `-1` for number of errors? I feel like it should be 0, but perhaps I'm missing something

**Comment by jonpas** [comment]
SQF Lint now has directory analyze option https://github.com/LordGolias/sqf/blob/92130853eadeec524ca641b2c4bde34f5fc18f52/sqflint.py with help of @PabstMirror 's script. It doesn't count the amount of warnings and errors as of now though.

File: tools/sqf_linter.py (hunk 7 above)
Comment (PabstMirror) [inline tools/sqf_linter.py]: -1 used to mean error, but I changed it, thanks for catching

**Pull request merged**